torchrl/modules/tensordict_module/actors.py

```diff
@@ -676,12 +676,22 @@ def _binary(self, value: torch.Tensor) -> torch.Tensor:
 
 
 def _process_action_space_spec(action_space, spec):
-    nest_action = False
+    original_spec = spec
+    composite_spec = False
     if isinstance(spec, CompositeSpec):
+        # this will break whenever our action is more complex than a single tensor
         try:
-            # this will break whenever our action is more complex than a single tensor
-            spec = spec["action"]
-            nest_action = True
+            spec_keys = spec.keys(True, True)
+            if "action" in spec_keys:
+                _key = "action"
+            else:
+                # the first key is the action
```

Comment:
getting the spec keys once and iterating twice can be dangerous is it's an iterable